lib/jsonb_accessor/macro.rb

```diff
@@ -44,19 +44,21 @@ def jsonb_accessor(jsonb_attribute, field_types)
           jsonb_values = public_send(jsonb_attribute) || {}
           jsonb_values.each do |store_key, value|
             name = names_and_store_keys.key(store_key)
-            write_attribute(name, value)
+            write_attribute(name, value) if name.present?
           end
           clear_changes_information if persisted?
         end
 
         # <jsonb_attribute>_where scope
-        scope("#{jsonb_attribute}_where", lambda do |attributes|
-          store_key_attributes = attributes.each_with_object({}) do |(name, value), new_attributes|
-            store_key = names_and_store_keys[name.to_s]
-            new_attributes[store_key] = value
-          end
-          jsonb_where(jsonb_attribute, store_key_attributes)
-        end)
+        unless respond_to? "#{jsonb_attribute}_where"
+          scope("#{jsonb_attribute}_where", lambda do |attributes|
+            store_key_attributes = attributes.each_with_object({}) do |(name, value), new_attributes|
+              store_key = names_and_store_keys[name.to_s]
```

Comment:
My concern here is that since we'll be accounting for store keys in the parent class, the child class won't be able to convert its store keys which could get kind of confusing. Although opting into store keys makes things a bit more complicated anyway.